src/main/java/cn/nukkit/level/GameRules.java

```diff
@@ -76,28 +80,24 @@ public void refresh() {
         stale = false;
     }
 
-    public void setGameRule(GameRule gameRule, boolean value) {
+    public <V> void setGameRule(GameRule gameRule, V value, Type type) {
         if (!gameRules.containsKey(gameRule)) {
             throw new IllegalArgumentException("Gamerule does not exist");
         }
-        gameRules.get(gameRule).setValue(value, Type.BOOLEAN);
+        ((Value<V>) gameRules.get(gameRule)).setValue(value, type);
         stale = true;
     }
 
+    public void setGameRule(GameRule gameRule, boolean value) {
+        setGameRule(gameRule, value, Type.BOOLEAN);
+    }
+
     public void setGameRule(GameRule gameRule, int value) {
-        if (!gameRules.containsKey(gameRule)) {
-            throw new IllegalArgumentException("Gamerule does not exist");
-        }
-        gameRules.get(gameRule).setValue(value, Type.INTEGER);
-        stale = true;
+        setGameRule(gameRule, value, Type.INTEGER);
     }
 
     public void setGameRule(GameRule gameRule, float value) {
-        if (!gameRules.containsKey(gameRule)) {
-            throw new IllegalArgumentException("Gamerule does not exist");
-        }
-        gameRules.get(gameRule).setValue(value, Type.FLOAT);
-        stale = true;
+        setGameRule(gameRule, value, Type.FLOAT);
     }
 
     public void setGameRules(GameRule gameRule, String value) throws IllegalArgumentException {
```

Comment:
rename to setGameRule

Reply:
it's already named setGameRule

Reply:
rename setGameRules(GameRule gameRule, String value) to setGameRule

Reply:
ok